server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/GetMailboxesMethod.java

```diff
@@ -102,17 +101,19 @@ private GetMailboxesResponse getMailboxesResponse(GetMailboxesRequest mailboxesR
         }
     }
 
-    private Stream<MailboxId> retrieveMailboxIds(Optional<ImmutableList<MailboxId>> mailboxIds, MailboxSession mailboxSession) throws MailboxException {
+    private Stream<Optional<Mailbox>> retrieveMailboxes(Optional<ImmutableList<MailboxId>> mailboxIds, MailboxSession mailboxSession) throws MailboxException {
```

Comment:
why do you want to change the return type ? Do we care about retrieving absent mailboxes ?